libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -375,24 +378,32 @@ internal class MapRouteLine(
             initPrimaryRoutePoints(routeFeatureData.first().route)
         }
 
-        val wayPointGeoJsonOptions = GeoJsonOptions().withMaxZoom(16)
```

Comment:
@Guardiola31337 do you have any background on why we've been using the `withMaxZoom` option here?

@tmpsantos do you think we should be explicitly narrowing down the zoom range for the runtime-generate tiles?

Reply:
That's a great question. The same is done in Maps `LayerSourceProvider`

```java
  @NonNull
  GeoJsonSource generateSource(Feature locationFeature) {
    return new GeoJsonSource(
      LOCATION_SOURCE,
      locationFeature,
      new GeoJsonOptions().withMaxZoom(16)
    );
  }
```

Maybe following same reasoning for consistency? 🤔 

I've noticed that this is done in a bunch of other places when creating the `GeoJsonOptions`, for instance `arrowShaftGeoJsonSource`, `arrowHeadGeoJsonSource`, `routeLineGeoJsonOptions` and `alternativeRouteLineGeoJsonOptions`.

Do you know @1ec5?

Reply:
@LukasPaczos `withMaxZoom(16)` could be used to 'mimic' our vector tiles. Afaik, our vector tileset's max zoom level is 16.

Geojson source with `GeoJsonOptions().withMaxZoom(16)` settings would create tiles of max zoom 16. For higher zoom values, tile would be overscaled.

Reply:
I assume there could be potential performance implications of not overzooming the runtime-added tiles for complex routes, right? Should we keep defaulting to z16 for out-of-the-box behavior and add an additional option to increase the max zoom level in case we have the map data available to match it?

Reply:
@LukasPaczos forgot to mention that native uses z18 as a maxZoom default value.

> I assume there could be potential performance implications of not overzooming the runtime-added tiles for complex routes, right?

Could be, but I wouldn't speculate before measuring actual impact. I think keeping both sources (geojson / vector) at max zoom level 16, would provide better visual results when geojson route is overlaid over basemap.

Reply:
Gotcha :+1: in this case, I'll leave the default to z16 and open it up for customization.